internal/pkg/describe/env.go

```diff
@@ -50,23 +64,70 @@ func NewEnvDescriber(projectName string, envName string) (*EnvDescriber, error)
 		return nil, fmt.Errorf("assuming role for environment %s: %w", env.ManagerRoleARN, err)
 	}
 	return &EnvDescriber{
-		env:          env,
-		store:        svc,
-		apps:         apps,
-		sessProvider: sess,
+		env:            env,
+		store:          svc,
+		proj:           proj,
+		apps:           apps,
+		rgClient:       resourcegroups.New(sess),
+		cloudformation: cloudformation.New(sess),
 	}, nil
 }
 
 // Describe returns info about a project's environment.
 func (e *EnvDescriber) Describe() (*EnvDescription, error) {
-	var tags map[string]string
+	appsForEnv, err := e.FilterAppsForEnv()
+	if err != nil {
+		return nil, err
+	}
+
 	return &EnvDescription{
 		Environment:  e.env,
-		Applications: e.apps,
-		Tags:         tags,
+		Applications: appsForEnv,
+		Tags:         e.proj.Tags,
 	}, nil
 }
 
+func (e *EnvDescriber) FilterAppsForEnv() ([]*archer.Application, error) {
+	var appObjects []*archer.Application
+
+	tags := map[string]string{
+		"ecs-environment": e.env.Name,
+	}
+	arns, err := e.rgClient.GetResourcesByTags(cloudformationResourceType, tags)
+	if err != nil {
+		return nil, err
+	}
+
+	var stacksOfEnvironment []string
+	for _, arn := range arns {
+		stack, err := e.parseARN(arn)
+		if err != nil {
+			return nil, err
+		}
+		stacksOfEnvironment = append(stacksOfEnvironment, stack)
+	}
+
+	for _, app := range e.apps {
+		stackName := stack.NameForApp(e.proj.Name, e.env.Name, app.Name)
+		for _, stack := range stacksOfEnvironment {
+			if stack == stackName {
+				appObjects = append(appObjects, app)
+			}
+		}
+	}
+
+	return appObjects, nil
+}
+
+func (e *EnvDescriber) parseARN(resourceArn string) (string, error) {
```

Comment:
what do you think moving this function to `resourcegroups` pkg so that it can be reused to parse the arn to get the stack name?

Ideally `GetResourcesByTags` can return a `resourceArn` type that has `stackName()` method that returns the stack name.

Reply:
Right now, cloudwatch has getAlarmName and codepipeline has getPipelineName that also call arn.Parse but these all do slightly different things with what's returned. Should I leave this here until GetResourcesByTags returns a `resourceArn` type and then we can move all of these parsing functions into `resourceGroups`? In the meantime, I'll rename this one to match the above.

Reply:
don't worry about that. Don't need to do that in this PR.